pkg/clients/release/releases.go

```diff
@@ -101,6 +104,42 @@ func (r *ReleaseController) GetReleases(namespace string) (*releaseApi.ReleaseLi
 	return releaseList, err
 }
 
+// StoreRelease stores a given Release as an artifact.
+func (r *ReleaseController) StoreRelease(release *releaseApi.Release) error {
+	artifacts := make(map[string][]byte)
+
+	releaseConditionStatus, err := r.GetReleaseConditionStatusMessages(release.Name, release.Namespace)
+	if err != nil {
+		return err
+	}
+	artifacts["release-condition-status-"+release.Name+".log"] = []byte(strings.Join(releaseConditionStatus, "\n"))
+
+	releaseYaml, err := yaml.Marshal(release)
+	if err != nil {
+		return err
+	}
+	artifacts["component-"+release.Name+".yaml"] = releaseYaml
+
+	if err := logs.StoreArtifacts(artifacts); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Get the message from the status of a release. Useful for debugging purposes.
+func (r *ReleaseController) GetReleaseConditionStatusMessages(name, namespace string) (messages []string, err error) {
+	release, err := r.GetRelease(name, "", namespace)
+	if err != nil {
+		return messages, fmt.Errorf("error getting Release: %v", err)
+	}
+	for _, condition := range release.Status.Conditions {
+		messages = append(messages, fmt.Sprintf("condition.Type: %s, condition.Status: %s, condition.Reason: %s\n",
+			condition.Type, condition.Status, condition.Reason))
+	}
+	return
```

Comment:
Shouldn't we return messages or something here? messages is modified on line 137 then nothing is done with it

Reply:
The function has defined the name of return value, then no need to add it again.

Reply:
Whoa, never knew that was possible. Very cool!